### PR TITLE
Increased test coverage

### LINKED ISSUE #116: Write new "custom" exceptions using requests
- [ ] Provide useful warning for lacking API url
- [ ] Provide useful error for lacking credentials (422 error on /auth/token
- [ ] Catch 403 errors for unauthenticated users

### LINKED ISSUE #126: Enable full functional testing of modeltestSDK
- [x] Set up mt database and api as part of sdk testing by gihub actions
- [x] Enable local testing towards locally hosted api, too
- [x] Write functional tests

### PR DESCRIPTION
**Pull request by snorrefjellvang**
closes #126 
closes #116

### CODE REVIEW

File: modeltestSDK/api.py [hunk 1]
```diff
@@ -275,20 +296,20 @@ def create(self, number: str, description: str, test_date: str, campaign_id: str
         body = dict(
             number=number,
             description=description,
-            type="Floater Test",
+            type=type,
```
Comment (EBGlom) [inline modeltestSDK/api.py:299]: we shouldn't be able to override the test type. FloaterTestAPI.create() should always create a floater test

File: modeltestSDK/api.py [hunk 2]
```diff
@@ -400,7 +401,7 @@ def create(self, number: str, description: str, test_date: str, campaign_id: str
         body = dict(
             number=number,
             description=description,
-            type="Wave Calibration",
+            type=type,
```
Comment (EBGlom) [inline modeltestSDK/api.py:404]: Same comment as for FloaterTest.create()

File: modeltestSDK/api.py [hunk 3]
```diff
@@ -522,7 +503,7 @@ def create(self, number: str, description: str, test_date: str, campaign_id: str
             number=number,
             description=description,
             test_date=test_date,
-            type="Wind Calibration",
+            type=type,
```
Comment (EBGlom) [inline modeltestSDK/api.py:506]: same comment as for FloaterTest.create()

File: modeltestSDK/api.py [hunk 4]
```diff
@@ -691,8 +655,8 @@ def get(self, filter_by: list = None, sort_by: list = None) -> Sensors:
         if sort_by is None:
             sort_by = list()
         params = create_query_parameters(filter_expressions=filter_by, sorting_expressions=sort_by)
-        data = self.client.get(self._resource_path, parameters=params)
-        return Sensors.parse_obj([dict(**item, client=self.client) for item in data])
+        data = self.client.get(self._resource_path, parameters=dict(**params, skip=skip, limit=limit))
+        return Sensors([parse_obj_as(Sensor, dict(**_, client=self.client)) for _ in data])
```
Comment (EBGlom) [inline modeltestSDK/api.py:659]: Use of "_" is normally limited to parameters not used. But does not cause any errors

File: modeltestSDK/resources.py [hunk 5]
```diff
@@ -2,15 +2,56 @@
...
+        if hasattr(self.client, self.__class__.__name__.lower()):
+            return getattr(self.client, self.__class__.__name__.lower())
+        elif hasattr(self.client, self.__class__.__name__[:-5].lower()):
+            return getattr(self.client, self.__class__.__name__[:-5].lower())
+        elif hasattr(self.client, self.__class__.__name__[:-7].lower()):
+            return getattr(self.client, self.__class__.__name__[:-7].lower())
+        elif hasattr(self.client, self.__class__.__name__[:-4].lower()):
+            return getattr(self.client, self.__class__.__name__[:-4].lower())
+        else:
+            return None
```
Comment (EBGlom) [inline modeltestSDK/resources.py:31]: should we instead add a equally named attribute inside each client object?

File: modeltestSDK/resources.py [hunk 6]
```diff
@@ -27,31 +68,42 @@ def to_pandas(self, **kwargs) -> pd.DataFrame:
...
+class Resources(List[Resource]):
+    def __init__(self, items: List[Resource] = None) -> None:
+        if items:
+            self._check_types(items)
+            super().__init__(items)
+
+    def _check_types(self, items: List[Resource]) -> None:
+        expected_types = self.__orig_bases__[0].__args__[0].__args__ \
+            if isinstance(self.__orig_bases__[0].__args__[0], typing._UnionGenericAlias) \
+            else [self.__orig_bases__[0].__args__[0]]
```
Comment (EBGlom) [inline modeltestSDK/resources.py:85]: pack this in separate method?

File: modeltestSDK/api.py (hunk 4 above)
Comment (snorrefjellvang) [reply modeltestSDK/api.py:659]: Replaced with "i"

File: modeltestSDK/resources.py (hunk 6 above)
Comment (snorrefjellvang) [reply modeltestSDK/resources.py:85]: Created separate method

**Review by EBGlom — APPROVED** [review]

**Pull request merged**